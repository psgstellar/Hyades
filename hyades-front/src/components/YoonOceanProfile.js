import React from "react";

// 테스팅 라이브러리는 한번씩 사용해보시기 위해서 현재 파일명과 컴포넌트 명, props 등을
// 자신의 id나 영문명 또는 nickname + Profile 형태로 바꿔주세요
// 그런다음 적절한 위치에 렌더링 해보시고, 테스트 라이브러리도 똑같이 바꿔주세요.
// 스프린트 1회차때는 이런 연습용 파일들이 삭제 될 예정이니, 필요한 예제코드는 가져가세요

const YoonOceanProfile = ({ username, name }) => {
  return (
    <div>
      <b>{username}</b>&nbsp;
      <span>({name})</span>
    </div>
  );
};

export default YoonOceanProfile;
