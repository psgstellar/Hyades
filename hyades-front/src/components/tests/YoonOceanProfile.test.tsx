// enzyme 와 @testing-library 는 같이 실행하면 에러가 납니다. 항상 주석 처리 잊지마세요 !
// 가끔 충돌 지점이 없어 실행될때도 있으나,,,,, 권장드리지 않습니다

import React from "react";
import YoonOceanProfile from "../YoonOceanProfile";
/*

// enzyme 적용시

import { shallow, mount, render } from "enzyme";
import { configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

configure({ adapter: new Adapter() });

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const wrapper = mount(<Profile username="yoonOcean" name="윤성연" />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders username and name", () => {
    const wrapper = mount(<Profile username="yoonOcean" name="윤성연" />);
    expect(wrapper.props().username).toBe("yoonOcean");
    expect(wrapper.props().name).toBe("윤성연");
  });
});

*/

// @testing-library 적용시

import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

afterEach(cleanup);
describe("<YoonOceanProfile />", () => {
  it("matches snapshot", () => {
    const utils = render(
      <YoonOceanProfile username="yoonOcean" name="윤성연" />
    );
    expect(utils.container).toMatchSnapshot();
  });
  it("shows the props correctly", () => {
    const utils = render(
      <YoonOceanProfile username="yoonOcean" name="윤성연" />
    );
    utils.getByText("yoonOcean"); //yoonOcean 라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText("(윤성연)"); // (윤성연) 이라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText(/윤/); // 정규식 /윤/ 을 통과하는 엘리먼트가 있는지 확인
  });
});
