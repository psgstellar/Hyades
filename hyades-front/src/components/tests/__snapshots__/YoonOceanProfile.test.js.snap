// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YoonOceanProfile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      yoonOcean
    </b>
     
    <span>
      (
      윤성연
      )
    </span>
  </div>
</div>
`;
